Python/Product/Analysis/LanguageServer/Server.cs

```diff
@@ -260,14 +280,17 @@ public sealed class Server : ServerBase, IDisposable {
             }
 
             await _analyzer.ReloadModulesAsync();
-
             // re-analyze all of the modules when we get a new set of modules loaded...
             foreach (var entry in _analyzer.ModulesByFilename) {
                 _queue.Enqueue(entry.Value.ProjectEntry, AnalysisPriority.Normal);
             }
         }
 
         public override Task<CompletionList> Completion(CompletionParams @params) {
+            IfTestWaitForAnalysisComplete();
```

Comment:
This is much better done in the test with one of the "Wait" methods. Why can't we do that?